product/src/controllers/categoriesController.js

```diff
@@ -0,0 +1,104 @@
+import categories from "../models/categories.js";
+
+class categoriesController {
+
+  static listarCategories = (req, res) => {
+    categories.find((err, Categories) => {
+      res.status(200).json(Categories)
+  })
+  }
+
+  static inserirCategories = (req, res) => {
+    let category = new categories(req.body);
+
+    category.save((err) => {
+
+      if(err) {
+        res.status(500).send({message: `${err.message} - falha ao cadastrar uma categoria.`})
+      } else {
+        res.status(201).send(category.toJSON())
+      }
+    })
+  }
+  //foi inserido uma categoria
+
+  static listarCategoriesPorId = (req, res) => {
+    const id = req.params.id;
+
+    categories.findById(id, (err, categories) => {
+      if(err) {
+        res.status(400).send({message: `${err.message} - Id da categoria não localizado.`})
+      } else {
+        res.status(200).send(categories);
+      }
+    })
+
+  //busquei by id
+
+
+
+  
+}
+
+
+
+static atualizarCategories = (req, res) => {
+  const id = req.params.id;
+
+  categories.findByIdAndUpdate(id, {$set: req.body}, (err) => {
+    if(!err) {
+      res.status(200).send({message: 'Categoria atualizada com sucesso'})
+    } else {
+      res.status(500).send({message: err.message})
+    }
+  })
+}
+
+static ativarCategories = (req, res) => {
+  const id = req.params.id;
+
+  categories.findByIdAndUpdate(id, {$set: {'STATUS':'ATIVA'}}, (err) => {
+    if(!err) {
+      res.status(200).send({message: 'Categoria atualizada com sucesso'})
+    } else {
+      res.status(500).send({message: err.message})
+    }
+  })
+}
+
+static excluirCategories = (req, res) => {
+  const id = req.params.id;
+
+  categories.findByIdAndDelete(id, (err) => {
+    if(!err){
+      res.status(200).send({message: 'Categoria removida com sucesso'})
```

Comment:
Quando o objeto não for retornado na resposta, pode ser enviado um código http 204

Reply:
então o "ativarCategoria" também deve retornar o status 204? já que apenas retornará uma mensagem?